cmd/gateway/nas/gateway-nas.go

```diff
@@ -118,15 +118,5 @@ func (g *NAS) Production() bool {
 
 // nasObjects implements gateway for Minio and S3 compatible object storage servers.
 type nasObjects struct {
-	*minio.FSObjects
-}
-
-// IsNotificationSupported returns whether notifications are applicable for this layer.
-func (l *nasObjects) IsNotificationSupported() bool {
-	return false
-}
-
-// IsCompressionSupported returns whether compression is applicable for this layer.
-func (l *nasObjects) IsCompressionSupported() bool {
-	return false
+	minio.ObjectLayer
```

Comment:
IsCompressionSupported() should still be false, if the gateways have different compression env settings the gateway would not behave consistently w.r.t compressed objects.

Reply:
The deployment of Gateways is always consistent and it should be. This is similar to access and secrets if they don't use the same settings they will fail to work. Same as encryption vault settings etc.

I would say we should enable it, compression is implemented at handler layer and there is no reason to disable it.

Reply:
Agree, perhaps this should be documented in NAS/ gateway docs. Approving this